soc/nxp/kinetis/ke1xz/Kconfig.defconfig

```diff
@@ -22,14 +22,10 @@ config NUM_IRQS
 config CPU_HAS_CUSTOM_FIXED_SOC_MPU_REGIONS
 	default y
 
-# This Kconfig is SOC Specific and configures the MCU at boot time
-# This SOC is configured via the following bits
-# 0000 0001 - Normal Boot
-# 0000 0100 - Enable Interrupts
-# 0000 1000 - Allow Reset detection from RESET_Pin
-# 0111 0000 - Reserved for Future Expansion
```

Comment:
Why remove the documentation here of how to use this option? Is it inaccurate?

Reply:
> Why remove the documentation here of how to use this option? Is it inaccurate?

To be consistent across all SOCs, I only document non-default/non-reset values.
Would you like to add documentation for default values ​​as well?

Reply:
Sure, doing it via DTS might make sense. I'm just asking because it seems like this would be useful information for the user.

Adding documentation to all SOCs would be nice, but I don't want to increase the scope of the PR. I just want to avoid removing useful comments like these in the name of consistency across SOCs.

Reply:
We shouldn't remove the documentation unless it is incorrect. If we want to be consistent then lets add a ticket to add this type of documentation to the other platforms.

Reply:
Do we need to add something to the migration guide about these values changing? I know this seems odd, but we are effectively changing the boot configuration for the KE and K8 parts, right? Currently the KE1XF would boot from ROM when BOOTCFG0 is asserted, and with this change it no longer will. I think changing the default is ok, but users may be confused if they are currently using this feature.

Reply:
I am not sure why this needs to be changed at all?

Reply:
> I am not sure why this needs to be changed at all?

The background:
- https://mcuoneclipse.com/2016/12/22/disabling-ezport-on-nxp-kinetis-to-solve-power-on-issues/
- https://mcuoneclipse.com/2020/06/09/disabling-nmi-non-maskable-interrupt-pin/
- https://community.nxp.com/t5/Kinetis-Microcontrollers/Kinetis-K02-reset-issue/m-p/1067426
- https://github.com/ARMmbed/mbed-os/issues/569
- https://community.nxp.com/t5/Kinetis-Microcontrollers/NMI-pin-floating/m-p/739809

Reply:
> I think changing the default is ok, but users may be confused if they are currently using this feature and the functionality changes without much warning

We may add this information to the migration guide or release notes, depending on which release it will be included in.